### PR TITLE
Update hasOwn tests to Jasmine

### PR DESCRIPTION
**Pull request by pedrosilva0111**
This migrates hasOwn tests from quint to jasmine for modernization.

### CODE REVIEW

File: test/hasOwn_test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+goog.module('datalayerhelper.plain.testing.hasOwn');
+goog.setTestOnly();
+
+const {hasOwn} = goog.require('plain');
+
+describe('the isPlainObject method', function() {
```
Comment (durbanie) [inline test/hasOwn_test.js:6]: Shouldn't this be the "hasOwn" function, not "isPlainObject"?

File: test/hasOwn_test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  /**
+   * Ensure that plain.hasOwn Determines if the value has a
```
Comment (durbanie) [inline test/hasOwn_test.js:8]: nit: don't capitalize "determines". I'd also rephrase this to something like: "Helper function used to ensure that plain.hasOwn...".

File: test/hasOwn_test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   * non-inherited property with the given key.
+   * @param {*} value The value to check.
+   * @param {string} key The property name.
+   * @param {boolean} expected If the value has a property named by key.
+   */
+  function assertHasOwn(value, key, expected) {
+    expect(hasOwn(value, key)).toBe(expected);
+  }
+
+  it('Assert that hasOwn recognizes simple type to not have properties', () => {
```
Comment (durbanie) [inline test/hasOwn_test.js:18]: I think you remove all of the "Assert that hasOwn". When the test runs, it will prefix each phrase with your describe statement. So if your describe above is "The hasOwn function", then (after my suggestion), the test would read:

"The hasOwn function recognizes simple type to not have own properties".

File: test/hasOwn_test.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  it('Assert that hasOwn recognizes Object properties', () => {
+    assertHasOwn({a: 1}, '0', false);
+    assertHasOwn({a: 1}, '1', false);
+    assertHasOwn({a: 1}, 'a', true);
+    assertHasOwn({a: 1}, 'valueOf', false);
+    assertHasOwn({a: 1}, 'constructor', false);
+  });
+
+  it('Assert that hasOwn recognizes function properties', () => {
+    const fn = function() {};
```
Comment (durbanie) [inline test/hasOwn_test.js:54]: I generally prefer arrow functions when assigning to variables, especially in nested functions: https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions

Comment (pedrosilva0111) [reply test/hasOwn_test.js]: Done, thanks!

File: test/hasOwn_test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+goog.module('datalayerhelper.plain.testing.hasOwn');
+goog.setTestOnly();
+
+const {hasOwn} = goog.require('plain');
+
+describe('the hasOwn method', function() {
+  /**
+   * Helper function used to insure that plain.hasOwn determines if
```
Comment (durbanie) [inline test/hasOwn_test.js:8]: nit: insure -> ensure
